CRM/Core/Payment.php

```diff
@@ -1144,6 +1144,7 @@ public static function handleIPN() {
         'processor_id' => @$_GET['processor_id'],
         'mode' => @$_GET['mode'],
         'q' => @$_GET['q'],
+        'task' => @$_GET['task'], // for Joomla
```

Comment:
Perhaps instead of hard-coding `q` and `task` params we should be using `$config->userFrameworkURLVar`?